src/server.c

```diff
@@ -585,7 +600,11 @@ qd_connection_t *qd_server_connection_impl(qd_server_t *server, qd_server_config
     ctx->connection_id = server->next_connection_id++;
     DEQ_INSERT_TAIL(server->conn_list, ctx);
     sys_mutex_unlock(&server->lock);
-    decorate_connection(ctx->server, ctx->pn_conn, config);
+    if (!!connector) {
+        ctx->connector = connector;
```

Comment:
Should the sys_atomic_inc(&connector->ref_count) done on line 1141 be moved here where the reference is set?